src/main/java/org/veupathdb/service/eda/ds/plugin/standalonemap/markers/MapBubbleSpecification.java

```diff
@@ -0,0 +1,49 @@
+package org.veupathdb.service.eda.ds.plugin.standalonemap.markers;
+
+import org.veupathdb.service.eda.generated.model.APIVariableDataShape;
+import org.veupathdb.service.eda.generated.model.CategoricalColorConfig;
+import org.veupathdb.service.eda.generated.model.ColorConfig;
+import org.veupathdb.service.eda.generated.model.ContinousColorConfig;
+import org.veupathdb.service.eda.generated.model.VariableSpec;
+
+import java.util.HashSet;
+import java.util.function.Function;
+import java.util.function.Supplier;
+
+import static org.veupathdb.service.eda.generated.model.OverlayType.CATEGORICAL;
+
+public class MapBubbleSpecification {
+  private final VariableSpec overlayVariable;
+  private OverlayRecoder overlayRecoder;
+  private Supplier<MarkerAggregator<Double>> aggregatorSupplier;
+
+  public MapBubbleSpecification(ColorConfig overlayConfig,
+                                Function<VariableSpec, String> varShapeFinder) {
+    this.overlayVariable = overlayConfig.getOverlayVariable();
+    if (CATEGORICAL.equals(overlayConfig.getOverlayType())) {
+
+      if (!varShapeFinder.apply(overlayVariable).equalsIgnoreCase(APIVariableDataShape.CATEGORICAL.getValue())) {
```

Comment:
this should probably allow ordinals as well. theyre a valid subcategory of categorical.